### PR TITLE
Optimized create_rank_to_trace_dict function

### PR DESCRIPTION
**Pull request by rzdao**
1. Improve create_rank_to_trace_dict function. Processing 10 compressed JSON files of 4.4m each used to take 17 seconds, but now it only takes 0.0005 seconds.
2. Variable name modified

### CODE REVIEW

File: hta/common/trace_file.py [hunk 1]
```diff
@@ -25,53 +26,49 @@ def create_rank_to_trace_dict(trace_path: str) -> Tuple[bool, Dict]:
...
+            if match:
+                rank = match.group().split(": ")[1]
+                rank_to_trace_dict[int(rank)] = file_path
+            else:
+                logger.error(
+                    "If the trace file does not have the rank specified in it, then add the following snippet "
+                    'key to the json files to use HTA; "distributedInfo": {"rank": 0}. If there are multiple '
+                    "traces files, then each file should have a unique rank value."
+                )
+                sys.exit()
```
Comment (fengxizhou) [inline hta/common/trace_file.py:70]: You may remove the sys.exit() because under certain scenarios not all files within trace_dir are trace files.

**Review by fengxizhou — APPROVED** [review]
Thanks for the optimization. It removes one of the speed bumps of the package.

File: hta/common/trace_file.py [hunk 2]
```diff
@@ -25,53 +25,52 @@ def create_rank_to_trace_dict(trace_path: str) -> Tuple[bool, Dict]:
...
+    ]
+    if len(file_list) == 0:
+        logger.warning(f"No trace file is found in {trace_dir}")
         return False, {}
 
     file_list = [
-        fn
-        for fn in os.listdir(trace_path)
-        if fn.endswith(".gz") or fn.endswith(".json")
+        fn for fn in os.listdir(trace_dir) if fn.endswith(".gz") or fn.endswith(".json")
```
Comment (briancoutinho) [inline hta/common/trace_file.py:40]: this looks repeat of line 32 to  36 no?

**Review by briancoutinho — APPROVED** [review]
"Processing 10 compressed JSON files of 4.4m each used to take 17 seconds, but now it only takes 0.0005 seconds."
thanks for adding, this is a really awesome speed up:)

**Pull request closed without merging**